daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -197,65 +141,71 @@ func handleTelioConfig(eventPath, deviceID, version string, prod bool, vpnLibCfg
 	} else {
 		telioConfig.Nurse = nil
 	}
-	return json.Marshal(telioConfig)
+
+	return &telioConfig, nil
+}
+
+type telioLoggerCb struct{}
+
+func (cb *telioLoggerCb) Log(logLevel teliogo.TelioLogLevel, payload string) *teliogo.TelioError {
+	log.Println(logLevelToPrefix(logLevel), "TELIO("+teliogo.GetVersionTag()+"): "+payload)
+	return nil
 }
 
 func New(prod bool, eventPath string, fwmark uint32,
-	vpnLibCfg vpn.LibConfigGetter, deviceID, appVersion string, eventsPublisher *vpn.Events) *Libtelio {
+	vpnLibCfg vpn.LibConfigGetter, deviceID, appVersion string, eventsPublisher *vpn.Events,
+) *Libtelio {
 	events := make(chan state)
-	logLevel := teliogo.TELIOLOGINFO
-
-	var telioConfigString string
-	cfg, err := handleTelioConfig(eventPath, deviceID, appVersion, prod, vpnLibCfg)
+	features, err := handleTelioConfig(eventPath, deviceID, appVersion, prod, vpnLibCfg)
 	if err != nil {
 		log.Println(internal.ErrorPrefix, "Failed to get telio config:", err)
 
-		defaultTelioConfig := &telioFeatures{}
-		defaultTelioConfig.Lana = &lanaConfig{
+		defaultTelioConfig := teliogo.GetDefaultFeatureConfig()
+		defaultTelioConfig.Lana = &teliogo.FeatureLana{
 			Prod:      prod,
 			EventPath: eventPath,
 		}
-		defaultTelioConfig.Direct = &directConfig{}
-		defaultTelioConfig.Nurse = &nurseConfig{
-			Fingerprint: deviceID,
+		defaultTelioConfig.Nurse = &teliogo.FeatureNurse{
+			Fingerprint:       deviceID,
+			HeartbeatInterval: defaultHeartbeatInterval,
 		}
 
-		fallbackTelioConfig, err := json.Marshal(defaultTelioConfig)
-		if err != nil {
-			log.Println(internal.ErrorPrefix, "Couldn't encode default telio config:", err)
-			fallbackTelioConfig = []byte(`{"direct":{}}`)
-		}
-		cfg = fallbackTelioConfig
+		features = &defaultTelioConfig
+	}
+
+	featuresString, err := json.Marshal(features)
+	if err != nil {
+		log.Println(internal.WarningPrefix, "failed to encode telio config:", err)
+		// pass through - encoding is for the logging purposes
 	}
-	telioConfigString = string(cfg)
+	log.Println(internal.InfoPrefix, "telio final config:", string(featuresString))
```

Comment:
maybe log this on `else`, otherwise `featuresString` might have invalid data if decoding fails?

Reply:
Yes, that makes sense. Fixed.